addons/addon-internal-auth-dep-check/packages/internal-auth-dep-check/lib/steps/internal-auth-dep-check-service.js

```diff
@@ -0,0 +1,255 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+const _ = require('lodash');
+const Service = require('@aws-ee/base-services-container/lib/service');
+
+const settingKeys = {
+  keyPairsTableName: 'dbKeyPairs',
+  usersTableName: 'dbUsers',
+  projectsTableName: 'dbProjects',
+  studiesTableName: 'dbStudies',
+  studyPermissionsTableName: 'dbStudyPermissions',
+  environemntsTableName: 'dbEnvironmentsSc',
+};
+
+class InternalAuthDepCheckService extends Service {
+  constructor() {
+    super();
+    this.dependency(['dbService']);
+  }
+
+  async init() {
+    await super.init();
+    // Setup services and SDK clients
+    this.dbService = await this.service('dbService');
+    this.dynamoDB = this.dbService.helper;
+  }
+
+  // method runs during pre deployment step
+  async execute() {
```

Comment:
For each of the error throwing lines here, could you store the list of uids and resource ids causing the error?
And then could you have one error summary in the end listing all uid-id combinations to be dealt with?

Example:
```
These are the resources blocking installation:

Make sure the following workspaces owned by internal users are terminated:
<Workspace ID 1> owned by user <uid> (<username>)
<Workspace ID 2> owned by user <uid> (<username>)

Make sure the following SSH keys owned by internal users are deactivated:
<SSH key name> owned by user <uid> (<username>)

...

```